lms/djangoapps/grades/rest_api/v1/gradebook_views.py

```diff
@@ -564,15 +565,41 @@ def get(self, request, course_key):
                     grade_conditions['effective_grade__lte'] = request.GET.get('assignment_grade_max')
                 if request.GET.get('assignment_grade_min'):
                     grade_conditions['effective_grade__gte'] = request.GET.get('assignment_grade_min')
-                annotations['selected_assignment_grade_in_range'] = Exists(
-                    subqueryset.filter(
+            if request.GET.get('course_grade_min') or request.GET.get('course_grade_max'):
+                grade_conditions = {}
+                if request.GET.get('course_grade_min'):
+                    course_grade_min = float(request.GET.get('course_grade_min')) / 100
+                    grade_conditions['percent_grade__gte'] = course_grade_min
+
+                if request.GET.get('course_grade_max'):
+                    course_grade_max = float(request.GET.get('course_grade_max')) / 100
+                    grade_conditions['percent_grade__lte'] = course_grade_max
+
+                subquery_grade_in_range = Exists(
+                    PersistentCourseGrade.objects.filter(
                         course_id=OuterRef('course'),
-                        user_id=OuterRef('user'),
-                        usage_key=UsageKey.from_string(request.GET.get('assignment')),
+                        user_id=OuterRef('user_id'),
                         **grade_conditions
                     )
                 )
-                q_objects.append(Q(selected_assignment_grade_in_range=True))
+                annotations['course_grade_in_range'] = subquery_grade_in_range
+                q_object = Q(course_grade_in_range=True)
+
+                if request.GET.get('course_grade_min'):
```

Comment:
This branch could be included in the `if request.GET.get('course_grade_min') ` branch above.

Reply:
I know what you mean. I did actually try to move it up there before, but I couldn't get it to work with combining the Q objects.


I had something like the following, but didn't know what to set `q_object` to. `None`? That doesn't work. 
```
            if request.GET.get('course_grade_min') or request.GET.get('course_grade_max'):
                grade_conditions = {}
                q_object = ???
                if request.GET.get('course_grade_min'):
                    course_grade_min = float(request.GET.get('course_grade_min')) / 100
                    grade_conditions['percent_grade__gte'] = course_grade_min

                    if course_grade_min == 0:
                        subquery_grade_absent = ~Exists(
                            PersistentCourseGrade.objects.filter(
                                course_id=OuterRef('course'),
                                user_id=OuterRef('user_id'),
                            )
                        )

                        annotations['course_grade_absent'] = subquery_grade_absent
                        q_object |= Q(course_grade_absent=True)

                if request.GET.get('course_grade_max'):
                    course_grade_max = float(request.GET.get('course_grade_max')) / 100
                    grade_conditions['percent_grade__lte'] = course_grade_max

                subquery_grade_in_range = Exists(
                    PersistentCourseGrade.objects.filter(
                        course_id=OuterRef('course'),
                        user_id=OuterRef('user_id'),
                        **grade_conditions
                    )
                )
                annotations['course_grade_in_range'] = subquery_grade_in_range
                q_object |= Q(course_grade_in_range=True)
                q_objects.append(q_object)
```

Reply:
Oh, I can just do `q_object = Q()`!